### PR TITLE
Yuliiaspryn/homework8

### PR DESCRIPTION
**Pull request by yspryn**

### CODE REVIEW

File: tests/yspryn/test_buildings_hw8.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+def test_buildings():
+    def json_dict(d):
+        return dict((k, list(v)) for k, v in d.items())
+
+    b = Building(1, 2, 2, 3)
+    b2 = Building(1, 2, 2, 3, 5)
+
+    assert json_dict(b.corners()) == {'north-east': [4, 4], 'south-east': [1, 4],
+                                      'south-west': [1, 2], 'north-west': [4, 2]}, "Corners"
```
Comment (kudinov-fedor) [inline tests/yspryn/test_buildings_hw8.py:12]: and what is return of      b.corners()   ?
from test it is not really clear

File: yspryn/hw7/caching.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+def decorator_answers(func):
+    remember_result = {}
+
+    def wrapper(n: int):
+        if n in remember_result:
+            return remember_result[n]
+        res = func(n)
+        remember_result[n] = res
+        return res
+    return wrapper
```
Comment (kudinov-fedor) [inline yspryn/hw7/caching.py:16]: try to reduce amount of returns and avoid nesting if possible:

`
def wrapper(n: int):
        if n not  in remember_result:
            remember_result[n] =  func(n)
        return remember_result[n]


`

File: yspryn/hw8/every_person.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        self.salary = salary
+        self.country = country
+        self.city = city
+        self.gender = gender
+
+    def name(self):
+        return f"{self.first_name} {self.last_name}"
+
+    def age(self):
+        today = date(2018, 1, 1)
```
Comment (kudinov-fedor) [inline yspryn/hw8/every_person.py:20]: better to move as constant on module level

```
TODAY = date(2018, 1, 1)

class Person:

    ...

   def age(self):
         birth_date = date(int(self.birth_date.split('.')[2]), int(self.birth_date.split('.')[1]), int(self.birth_date.split('.')[0]))
        age = TODAY .year - birth_date.year - ((TODAY .month, today.day) < (birth_date.month, birth_date.day))
        return age


```

File: tests/yspryn/test_every_person_hw8.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+
+def test_every_person():
+    p1 = Person("John", "Smith", "19.09.1979", "welder", 15, 3600, "Canada", "Vancouver", "male")
+    p2 = Person("Hanna Rose", "May", "05.12.1995", "designer", 2.2, 2150, "Austria", "Vienna")
+    assert p1.name() == "John Smith", "Name"
+    assert p1.age() == 38, "Age"
+    assert p2.work() == "Is a designer", "Job"
+    assert p1.money() == "648 000", "Money"
+    assert p2.home() == "Lives in Vienna, Austria", "Home"
```
Comment (kudinov-fedor) [inline tests/yspryn/test_every_person_hw8.py:11]: missing tests, not all logic is covered

File: yspryn/hw8/three_chefs.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    def add_food(self, food_count, food_price):
+        self.food_order.append((food_count, food_price))
+
+    def add_drink(self, drink_count, drink_price):
+        self.drink_order.append((drink_count, drink_price))
+
+    def total(self):
+        total_food = 0
+        for item, price in self.food_order:
+            total_food += item * price
```
Comment (kudinov-fedor) [inline yspryn/hw8/three_chefs.py:18]: try sum   comprehension

**Review by kudinov-fedor — DISMISSED** [review]

File: yspryn/hw8/every_person.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        age = Today.year - birth_date.year - ((Today.month, Today.day) < (birth_date.month, birth_date.day))
+        return age
+
+    def work(self):
+        if self.gender is "male":
+            return f"He is a {self.job}"
+        elif self.gender is "female":
+            return f"She is a {self.job}"
+        else:
+            return f"Is a {self.job}"
```
Comment (fkudinov-mn) [inline yspryn/hw8/every_person.py:34]: avoid multiple returns

File: yspryn/hw8/every_person.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        self.city = city
+        self.gender = gender
+
+    def name(self):
+        return f"{self.first_name} {self.last_name}"
+
+    def age(self):
+        # today = date(2018, 1, 1)
+        birth_date = date(int(self.birth_date.split('.')[2]), int(self.birth_date.split('.')[1]),
+                          int(self.birth_date.split('.')[0]))
```
Comment (fkudinov-mn) [inline yspryn/hw8/every_person.py:24]: there is a way to convert string to date based on pattern

**Review by kudinov-fedor — APPROVED** [review]

**Review by fkudinov-mn — APPROVED** [review]

**Review by kudinov-fedor — DISMISSED** [review]

**Pull request merged**